samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Activities/ContentActivities.cs

```diff
@@ -0,0 +1,360 @@
+using System.Threading.Tasks;
+
+namespace AgentChainingSample.Activities;
+
+/// <summary>
+/// Result returned by the AssembleFinalArticleActivity
+/// </summary>
+public class ArticleResult
+{
+    /// <summary>
+    /// The complete HTML content of the article
+    /// </summary>
+    public string HtmlContent { get; set; } = string.Empty;
+    
+    /// <summary>
+    /// The local file path where the HTML content is saved
+    /// </summary>
+    public string FilePath { get; set; } = string.Empty;
+    
+    /// <summary>
+    /// The URL to the article in blob storage (kept for compatibility, always empty)
+    /// </summary>
+    public string BlobUrl { get; set; } = string.Empty;
+}
+
+/// <summary>
+/// Activity to research a topic using the research agent with web search
+/// </summary>
+[DurableTask]
+public class ResearchTopicActivity : TaskActivity<string, ResearchData>
+{
+    private readonly ResearchAgentService _researchAgentService;
+    private readonly ILogger<ResearchTopicActivity> _logger;
+
+    public ResearchTopicActivity(ResearchAgentService researchAgentService, ILogger<ResearchTopicActivity> logger)
+    {
+        _researchAgentService = researchAgentService;
+        _logger = logger;
+    }
+
+    public override async Task<ResearchData> RunAsync(TaskActivityContext context, string topic)
+    {
+        _logger.LogInformation("Researching topic using web search: {Topic}", topic);
+
+        try
+        {
+            string researchJson = await _researchAgentService.ResearchTopicAsync(topic);
+            _logger.LogInformation("Successfully collected research data for topic: {Topic}", topic);
+            
+            // Clean the JSON response
+            string cleanJson = _researchAgentService.CleanJsonResponse(researchJson);
+            
+            // Parse the JSON into research data
+            ResearchData researchData = ResearchData.FromJson(cleanJson);
+            return researchData;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error researching topic: {Topic}", topic);
+            throw;
+        }
+    }
+}
+
+/// <summary>
+/// Activity to create article content using the content generation agent with knowledge files
+/// </summary>
+[DurableTask]
+public class CreateArticleActivity : TaskActivity<(string topic, ResearchData researchData), string>
+{
+    private readonly ContentGenerationAgentService _contentGenerationService;
+    private readonly ILogger<CreateArticleActivity> _logger;
+
+    public CreateArticleActivity(ContentGenerationAgentService contentGenerationService, ILogger<CreateArticleActivity> logger)
+    {
+        _contentGenerationService = contentGenerationService;
+        _logger = logger;
+    }
+
+    public override async Task<string> RunAsync(TaskActivityContext context, (string topic, ResearchData researchData) input)
+    {
+        _logger.LogInformation("Creating article for topic: {Topic}", input.topic);
```

Comment:
Nit: use PascalCase in named tuples as it aligns with property naming guidelines.